File: TypedActorExample/src/main/scala/org/akka/essentials/calculator/example3/ChildActor.scala
package org.akka.essentials.calculator.example3
import akka.actor.actorRef2Scala
import akka.actor.Actor
import akka.event.Logging
import akka.actor.ActorLogging

class ChildActor extends Actor with ActorLogging {

  override def preStart() {
    log.info("Child Actor Started > {}", self.path)
  }

  def receive: Receive = {
    case message: String => throw new IllegalArgumentException("boom!")
    case message: Integer => sender ! message * message
    case _ => unhandled()
  }

  override def postStop() {
    log.info("Child Actor Stopped > {}", self.path)
  }
}

File: TypedActorExample/src/main/java/org/akka/essentials/calculator/example3/ChildActor.java
package org.akka.essentials.calculator.example3;

import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class ChildActor extends UntypedActor {

	LoggingAdapter log = Logging.getLogger(getContext().system(), this);

	public void preStart() {
		log.info("Child Actor Started > {}", self().path());
	}

	@Override
	public void onReceive(Object message) throws Exception {

		if (message instanceof String) {
			throw new IllegalArgumentException("boom!");
		} else if (message instanceof Integer) {
			Integer value = (Integer) message;
			getSender().tell(value * value);
		} else
			unhandled(message);

	}

	public void postStop() {
		log.info("Child Actor Stopped > {}", self().path());
	}

}
